mitodl/mitxpro
# Jk/hubspot ecommerce bridge

#### Pre-Flight checklist


#### What are the relevant tickets?
#18 #276 

#### What's this PR do?
Expands the hubspot sync settings and implements the tasks to sync Orders, Lines, and Products

#### How should this be manually tested?
Run the hubspot configuration management command:
`docker-compose run web ./manage.py configure_hubspot_bridge`

In the shell, use the hubspot serializers to serialize objects and use the hubspot tasks `sync_<OBJECT>_with_hubspot(<OBJECT_ID>)` to send the sync messages.

Check for errors with the hubspot api function `get_sync_errors()` and `get_sync_status(<CONTACT/PRODUCT/DEAL/LINE_ITEM>, <OBJECT_ID>)`

#### Where should the reviewer start?
`hubspot/api`

#### Any background context you want to provide?
The synced objects are not all currently appearing in the hubspot app. I'm in communication with hubspot about this, but the sync messages appear to be being received and sync statuses are logged.

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/3o7TKpkZRUTt5wuwuc/giphy-downsized.gif)
